### PR TITLE
Add meta information to compositions

### PR DESCRIPTION
**Pull request by zugao**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.

### CODE REVIEW

File: component/common.libsonnet [hunk 1]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/offered': 'true',
+      'metadata.appcat.vshn.io/serviceID': 'vshn-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaSOS(provider) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': provider + '.ch ObjectBucket',
```
Comment (glrf) [inline component/common.libsonnet:58]: I think technically it's `Exoscale` (without `.ch`) and `cloudscale.ch` (with `.ch`) , but :shrug:

**Review by glrf — APPROVED** [review]
LGTM

File: component/common.libsonnet (hunk 1 above)
Comment (tobru) [reply component/common.libsonnet:58]: Yes, exactly. We should always use the spelling as asked by the companies. (Like we write VSHN, not Vshn)

Comment (zugao) [reply component/common.libsonnet:58]: It's a bug :) I will fix it.

File: component/common.libsonnet [hunk 2]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/serviceID': 'exoscale-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaVshn(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'VSHN Managed ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by VSHN',
```
Comment (tobru) [inline component/common.libsonnet:42]:
```suggestion
      'metadata.appcat.vshn.io/description': dbname + ' instances by VSHN',
```

Please follow the naming from https://products.docs.vshn.ch/products/appcat/postgresql.html

File: component/common.libsonnet [hunk 3]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
   },
 };
 
+local vshnMetaDBaaSExoscale(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'Exoscale ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by Exoscale',
```
Comment (tobru) [inline component/common.libsonnet:26]:
```suggestion
      'metadata.appcat.vshn.io/description': dbname + ' DBaaS instances by Exoscale',
```

File: component/common.libsonnet [hunk 4]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+};
+
+local vshnMetaVshn(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'VSHN Managed ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by VSHN',
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/vshn-dbaas/' + std.asciiLower(dbname) + '/create.html',
+      'metadata.appcat.vshn.io/zone': 'local cluster',
+      'metadata.appcat.vshn.io/flavor': 'standalone, clustered',
```
Comment (tobru) [inline component/common.libsonnet:45]: Shouldn't we make that configurable?

File: component/common.libsonnet [hunk 5]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+  },
+};
+
+local vshnMetaVshn(dbname) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': 'VSHN Managed ' + dbname,
+      'metadata.appcat.vshn.io/description': dbname + ' instances hosted by VSHN',
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/vshn-dbaas/' + std.asciiLower(dbname) + '/create.html',
+      'metadata.appcat.vshn.io/zone': 'local cluster',
```
Comment (tobru) [inline component/common.libsonnet:44]:
```suggestion
      'metadata.appcat.vshn.io/zone': 'same as APPUiO Cloud zone',
```

Not sure about that...

File: component/common.libsonnet [hunk 6]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/product-description': 'https://products.docs.vshn.ch/products/appcat/' + std.asciiLower(dbname) + '.html',
+    },
+    labels+: {
+      'metadata.appcat.vshn.io/offered': 'true',
+      'metadata.appcat.vshn.io/serviceID': 'vshn-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaSOS(provider) = {
```
Comment (tobru) [inline component/common.libsonnet:55]:
```suggestion
local vshnMetaObjectStorage(provider) = {
```

IMHO SOS is the product name from Exoscale

File: component/common.libsonnet [hunk 7]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/serviceID': std.asciiLower(std.rstripChars(provider, '.ch')) + '-objectbucket',
+    },
+  },
+};
+
 {
   SyncOptions: syncOptions,
+  VshnMetaDBaaSExoscale(dbname):
+    vshnMetaDBaaSExoscale(dbname),
+  VshnMetaSOS(provider):
```
Comment (tobru) [inline component/common.libsonnet:76]:
```suggestion
  VshnMetaObjectStorage(provider):
```

File: component/common.libsonnet [hunk 8]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/offered': 'true',
+      'metadata.appcat.vshn.io/serviceID': 'vshn-' + std.asciiLower(dbname),
+    },
+  },
+};
+
+local vshnMetaSOS(provider) = {
+  metadata+: {
+    annotations+: {
+      'metadata.appcat.vshn.io/displayname': provider + ' ObjectBucket',
```
Comment (tobru) [inline component/common.libsonnet:58]:
```suggestion
      'metadata.appcat.vshn.io/displayname': provider + ' Object Storage',
```

This is the official name of the service: https://products.docs.vshn.ch/products/appcat/objectstorage.html

File: component/common.libsonnet [hunk 9]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/description': 'S3 compatible object bucket hosted by ' + provider,
```
Comment (tobru) [inline component/common.libsonnet:59]:
```suggestion
      'metadata.appcat.vshn.io/description': 'S3 compatible object storage hosted by ' + provider,
```

ObjectBucket (object bucket) is our invention afaik and not an official / widely known term.

File: component/common.libsonnet [hunk 10]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+      'metadata.appcat.vshn.io/end-user-docs-url': 'https://docs.appuio.cloud/appcat/object-storage/create.html',
+      'metadata.appcat.vshn.io/zone': provider + ' zones: ' +
+                                      if provider == 'Exoscale' then 'de-fra-1, de-muc-1, at-vie-1, ch-gva-2, ch-dk-2, bg-sof-1' else 'lpg and rma',
```
Comment (tobru) [inline component/common.libsonnet:62]: I don't really like this hardcoded list of zones, at least make it a variable to be reused per provider. Soon(ish) we'll provide the service also on other cloud providers, and then we're already better prepared.

File: component/common.libsonnet [hunk 11]
```diff
@@ -19,6 +19,62 @@ local syncOptions = {
...
+    },
+  },
+};
+
 {
   SyncOptions: syncOptions,
+  VshnMetaDBaaSExoscale(dbname):
+    vshnMetaDBaaSExoscale(dbname),
+  VshnMetaSOS(provider):
+    vshnMetaSOS(provider),
```
Comment (tobru) [inline component/common.libsonnet:77]:
```suggestion
    vshnMetaObjectStorage(provider),
```

File: component/objectstorage.jsonnet [hunk 12]
```diff
@@ -69,7 +69,10 @@ local compositionCloudscale =
     },
   };
 
-  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'cloudscale.objectbuckets.appcat.vshn.io') + common.SyncOptions + {
+  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'cloudscale.objectbuckets.appcat.vshn.io') +
+  common.SyncOptions +
+  common.VshnMetaSOS('cloudscale.ch') +
```
Comment (tobru) [inline component/objectstorage.jsonnet:74]:
```suggestion
  common.VshnMetaObjectStorage('cloudscale.ch') +
```

File: component/objectstorage.jsonnet [hunk 13]
```diff
@@ -187,7 +190,10 @@ local compositionExoscale =
     },
   };
 
-  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'exoscale.objectbuckets.appcat.vshn.io') + common.SyncOptions + {
+  kube._Object('apiextensions.crossplane.io/v1', 'Composition', 'exoscale.objectbuckets.appcat.vshn.io') +
+  common.SyncOptions +
+  common.VshnMetaSOS('Exoscale') +
```
Comment (tobru) [inline component/objectstorage.jsonnet:195]:
```suggestion
  common.VshnMetaObjectStorage('Exoscale') +
```

**Review by tobru — CHANGES_REQUESTED** [review]

File: component/common.libsonnet (hunk 5 above)
Comment (zugao) [reply component/common.libsonnet:44]: @Kidswiss what's your take on that? I would go with `APPUiO Cloud zone`

Comment (Kidswiss) [reply component/common.libsonnet:44]: Isn't there a fact we could use to set this value?

File: component/common.libsonnet (hunk 4 above)
Comment (tobru) [reply component/common.libsonnet:45]: This assumes that all services are available in all flavors right away and gives no way to have services, e.g. only in standalone flavor.

File: component/common.libsonnet (hunk 10 above)
Comment (zugao) [reply component/common.libsonnet:62]: I defined an array for each one of them.

File: component/common.libsonnet (hunk 4 above)
Comment (zugao) [reply component/common.libsonnet:45]: I updated it to have only standalone version for that. Also added a second argument to the function.

File: component/common.libsonnet (hunk 5 above)
Comment (zugao) [reply component/common.libsonnet:44]: we could use region from the [facts]( https://syn.tools/commodore/reference/parameters.html#_facts)

Comment (zugao) [reply component/common.libsonnet]: I added region from facts in the zone annotation.

**Review by tobru — APPROVED** [review]

**Pull request merged**